python/cuml/cuml/manifold/umap.pyx

```diff
@@ -533,6 +514,21 @@ class UMAP(UniversalBase,
             else:
                 umap_params.p = <float>self.metric_kwds.get('p')
 
+            if self.build_algo == "brute_force_knn":
+                umap_params.build_algo = graph_build_algo.BRUTE_FORCE_KNN
+            else:
+                umap_params.build_algo = graph_build_algo.NN_DESCENT
+                build_kwds = self.build_kwds or {}
+                umap_params.nn_descent_params.graph_degree = <uint64_t> build_kwds.get("nnd_graph_degree", 64)
+                umap_params.nn_descent_params.intermediate_graph_degree = <uint64_t> build_kwds.get("nnd_intermediate_graph_degree", 128)
+                umap_params.nn_descent_params.max_iterations = <uint64_t> build_kwds.get("nnd_max_iterations", 20)
+                umap_params.nn_descent_params.termination_threshold = <float> build_kwds.get("nnd_termination_threshold", 0.0001)
+                umap_params.nn_descent_params.return_distances = <bool> build_kwds.get("nnd_return_distances", True)
+                umap_params.nn_descent_params.n_clusters = <uint64_t> build_kwds.get("nnd_n_clusters", 1)
+                # Forward metric & metric_kwds to nn_descent
+                umap_params.nn_descent_params.metric = <RaftDistanceType> umap_params.metric
+                umap_params.nn_descent_params.metric_arg = umap_params.p
```

Comment:
The actual fix is here (plumbing through the metric options to `nn_descent_params`). In the long run we should redesign the C++ layer to remove duplicate options - for now just ensuring they're forwarded correctly seems sufficient.

Everything else here is a simplification of the current pre-existing code.